fsleyes_plugin_shimming_toolbox/st_plugin.py

```diff
@@ -48,163 +48,109 @@
-        with wx.MessageDialog(
-                self,
-                message,
-                caption=caption,
-                style=wx.OK | wx.CENTRE,
-                pos=wx.DefaultPosition,
-        ) as msg:
-            msg.ShowModal()
-
-    def verify_version(self):
-        """Check if the plugin version is the same as the one in the shimming-toolbox directory."""
-
-        st_path = os.path.realpath(__file__)
-        plugin_file = os.path.join(st_path, "gui", "st_plugin.py")
-
-        plugin_file_exists = os.path.isfile(plugin_file)
-
-        if not plugin_file_exists:
-            return
-
-        # Check the version of the plugin
-        with open(plugin_file) as plugin_file_reader:
-            plugin_file_lines = plugin_file_reader.readlines()
-
-        plugin_file_lines = [x.strip() for x in plugin_file_lines]
-        version_line = f'VERSION = "{VERSION}"'
-        plugin_is_up_to_date = True
-        version_found = False
-
-        for lines in plugin_file_lines:
-            if lines.startswith("VERSION = "):
-                version_found = True
-                if not lines == version_line:
-                    plugin_is_up_to_date = False
-
-        if version_found is False or plugin_is_up_to_date is False:
-            message = """
-                A more recent version of the ShimmingToolbox plugin was found in your
-                ShimmingToolbox installation folder. You will need to replace the current
-                FSLeyes plugin with the new one.
-                To proceed, go to: File -> Load plugin -> st_plugin.py. Then, restart FSLeyes.
-            """
-            self.show_message(message, "Warning")
-        return
-
+    # The CanvasPanel view is used for most FSLeyes plugins so we decided to stick to it
     @staticmethod
     def supportedViews():
-        """I am not sure what this method does."""
-        from fsleyes.views.orthopanel import OrthoPanel
-
-        return [OrthoPanel]
+        return [canvaspanel.CanvasPanel]
 
     @staticmethod
     def defaultLayout():
-        """This method makes the control panel appear on the bottom of the FSLeyes window."""
-        return {
-            "location": wx.BOTTOM,
-            "title": "Shimming Toolbox"
-        }
+        """This method makes the control panel appear on the top of the FSLeyes window."""
+        return {"location": wx.TOP, "title": "Shimming Toolbox"}
 
+    def __init__(self, parent, overlayList, displayCtx, ctrlPanel):
+        """Initialize the control panel.
 
-class TabPanel(wx.ScrolledWindow):
-    def __init__(self, parent):
-        super().__init__(parent=parent)
+        Generates the widgets and adds them to the panel.
 
+        """
+        super().__init__(parent, overlayList, displayCtx, ctrlPanel)
+        # Create a notebook to navigate between the different functions.
         nb = wx.Notebook(self)
+
+        # Add a terminal to log messages
+        nb.terminal_component = Terminal(self)
```

Comment:
The notebook does not have a `terminal_component`, we should wrap the notebook in a class if we want to add an attribute to it. I'll implement a version in the next commit

Reply:
Added here: https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox/pull/40/commits/4631321f4567bb8d9ae85303c28252034c49f241